src/main/java/net/consensys/orion/storage/QueryPrivacyGroupStorage.java

```diff
@@ -75,7 +75,7 @@ public String generateDigest(QueryPrivacyGroupPayload data) {
       if (result.isPresent()) {
         if (data.isToDelete()) {
           result.get().privacyGroupId().remove(data.privacyGroupToAppend());
-        } else {
+        } else if (!result.get().privacyGroupId().contains(data.privacyGroupToAppend())) {
```

Comment:
It might make this easier to parse if the "result.isPresent()" section is broken out to a separate function which takes a QueryPrivacyGroupPayload, and probably the same thing ...